Assignment4/rarycops-a180061/task07.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1tV5j-DRcpPtoJGoMj8v2DSqR_9wyXeiE
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2020-2021/master/Assignment4"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/resources/example6.rdf", format="xml")
+
+#RDFLib
+for s, p, o in g.triplets((None, RDFS.subClassOf, ns.Person)):
+    for s, p, o in g.triplets((None, RDFS.subClassOf, s)):
+        print(s)
+    for s, p, o in g.triplets((None, RDF.type,s)):
+        print(s)
+        for s, p, o in g.triplets((s, None, None)):
+            print(s,p,o)
+    for s, p, o in g.triplets((None, RDF.type,ns.Person)):
+        print(s)
+        for s, p, o in g.triplets((s, None, None)):
+            print(s,p,o)
```

Comment:
The last iteration is quite confusing because you are binding and returning the same variable name. I recommend separating it. Also you don't seem to be using the first filter for the third iteration